scripts/dev/infra-sync.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# Fetches cluster artifacts from infra.rox.systems and merged them into a kubeconfig file.
+
+set -euo pipefail
+
+os=$(uname -o)
+infra_dir="${HOME}/.kube/infra"
+clusters_dir="${infra_dir}/clusters"
+infra_kubeconfig="${INFRA_KUBECONFIG:-${infra_dir}/kubeconfig}"
+extra_kubeconfigs="${EXTRA_KUBECONFIGS:-}"
+
+delete_cluster_artifacts() {
+    echo "Deleting old cluster artifacts..."
+    # Just a couple of safety checks to make sure that we are not attempting to delete
+    # something like `/`.
+    mkdir -p "${clusters_dir}"
+    if [[ ( "${#clusters_dir}" -le 1 ) || ( ! -d "${clusters_dir}" ) ]]; then
+        echo >&2 "The clusters directory '${clusters_dir}' does not seem properly initialized."
+        exit 1
+    fi
+    # A call of the form `rm -rf some_directory/*` would fail if `some_directory/*` doesn't
+    # evaluate to anything.
+    find "${clusters_dir}" -mindepth 1 -maxdepth 1 -type d -exec rm -rf {} +
+}
+
+export INFRA_TOKEN="${INFRA_TOKEN:-}"
+if [[ "${INFRA_TOKEN}" == "" ]]; then
+    case "${os}" in
+        "Darwin")
+            if token=$(security find-generic-password -s infra.rox.systems -D token -a "" -l acs -w); then
+                INFRA_TOKEN="${token}"
+            else
+                echo >&2 "Failed to retrieve token from Keychain"
+                echo >&2 "Use the following command for saving your infra token in Keychain:"
+                echo >&2
+                echo >&2 "  $ security add-generic-password -s infra.rox.systems -D token -a \"\" -l acs -T \"\" -w \"YOUR_INFRA_TOKEN\""
+                echo >&2
+                exit 1
+            fi
+            ;;
+        *)
+            echo >&2 "INFRA_TOKEN is not set and support for retrieving a token has not been implemented on ${os}."
+            exit 1
+            ;;
+    esac
+fi
+
+echo "Syncing your infra clusters..."
+delete_cluster_artifacts
+
+# Fetching artifacts for infra clusters.
+cluster_names=""
+if ! cluster_names=$(infractl list --json | jq -r '(.Clusters // [])[] | select(.Status==2).ID'); then
+    echo >&2 "Failed to retrieve clusters names from infra."
+    exit 1
+fi
+
+idx=0
+for cluster_name in ${cluster_names}; do
+    idx=$((idx + 1))
+    cluster_dir="${clusters_dir}/${cluster_name}"
+    echo "Downloading artifacts for cluster ${cluster_name} into ${clusters_dir}"
+    infractl artifacts "${cluster_name}" --download-dir="${cluster_dir}" >/dev/null
+    kubeconfig="${cluster_dir}/kubeconfig"
+    chmod 600 "${kubeconfig}"
+    echo "Stored kubeconfig at ${kubeconfig}"
+done
+
+if [[ $idx == 0 ]]; then
+    echo "No ready clusters found."
+fi
+
+kubecfg_merge() {
+    # Backup previous kubeconfig file.
+    infra_kubeconfig_backup="${infra_kubeconfig}.bkp"
```

Comment:
I agree with @tommartensen - I don't think we should overwrite the original kubeconfig, at least not without any other option.
If I run Minikube/Minishift/k3s/... locally, I would loose the config for that, right?
Sure, I could always copy that over from the backup by hand, but I would need to do so every time after I run this script.
I'd either lean towards a prompt asking the user what to do, and/or a flag to set a behaviour (append vs overwrite).
WDYT?

Reply:
With my recent changes I have changed the script so that it doesn't touch the `~/.kube/config` by default. Instead, a different kubeconfig is used by default, named `~/.kube/infra/kubeconfig`. This file is maintained by the script and overwritten on any invocation. You *can* point the script to a different file by setting `INFRA_KUBECONFIG`, though.

I guess this is a reasonably safe resolution?

Reply:
Yes, that is great! Thank you for improving